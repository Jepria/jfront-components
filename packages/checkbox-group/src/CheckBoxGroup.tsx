import React, { useState, useEffect, ReactNode } from "react"
import styled from "styled-components"
import { Label } from "@jfront/ui-label"
import {
  LoadingImage as Loading,
  ExclamationImage as Exclamation,
} from "@jfront/ui-icons"

interface CheckBoxGroupInterface {
  children: ReactNode[]
  name?: string
  /**
   * use values to external state control
   */
  values?: any[]
  text?: string
  error?: string
  disabled?: boolean
  isLoading?: boolean
  style?: React.CSSProperties
  className?: string
  /**
   * Обработчик изменения значения 'checked' одного из дочерних элементов
   */
  onChange?: (
    name?: string,
    value?: any[],
    event?: React.ChangeEvent<any>,
  ) => void
}

const StyledCheckBoxGroup = styled.div`
  display: flex;
  width: 100%;
  justify-content: left;
`

interface StyledUlProps {
  error?: string
}

const StyledUl = styled.div<StyledUlProps>`
  margin: 2px;
  padding: 5px;
  font-family: tahoma, arial, helvetica, sans-serif;
  font-size: 12px;
  border: 1px solid grey;
  padding-left: 0;
  ${(props) => (props.error ? "border: 1px solid red;" : "")};
`
const LoadingImage = styled(Loading)`
  margin-top: 5px;
`

const ExclamationImage = styled(Exclamation)`
  margin-top: 5px;
`

export const CheckBoxGroup = React.forwardRef<
  HTMLDivElement,
  CheckBoxGroupInterface
>(
  (
    {
      children,
      name,
      values,
      text,
      disabled,
      isLoading,
      error,
      style,
      className,
      onChange,
    },
    ref,
  ) => {
    const [state, setState] = useState<any[]>([])

    useEffect(() => {
      if (values) {
        setState(values)
      }
    }, [values])

    const handleCheckboxChange = (
      _value: React.ReactText,
      event: React.ChangeEvent<any> | undefined,
    ) => {
      const newValue = values ? values.slice() : state.slice()

      const changedValueIndex = newValue.findIndex(
        (stateValue) => stateValue === _value,
      )

      if (event && event.target.checked) {
        newValue.push(_value)
      } else {
        newValue.splice(changedValueIndex, 1)
      }

      setState(newValue)

      if (onChange) {
        onChange(name, newValue, event)
      }
    }

    return (
      <StyledCheckBoxGroup>
        {text && <Label>{text}</Label>}
        <StyledUl style={style} className={className} ref={ref} error={error}>
          {React.Children.map(children, (checkbox) => {
            if (!React.isValidElement(checkbox)) {
              return null
            }

            const isFound = state.find(
              (stateValue) => stateValue === checkbox.props.value,
            )

            const checked = undefined !== isFound

            return React.cloneElement(checkbox, {
              disabled: checkbox.props.disabled || disabled,
              checked: checked,
              onChange:
                checkbox.props.onChange === undefined
                  ? (event: any, _text: any) =>
                      handleCheckboxChange(checkbox.props.value, event)
                  : checkbox.props.onChange,
            })
          })}
        </StyledUl>
        {isLoading && <LoadingImage />}
        {error !== undefined && <ExclamationImage title={error} />}
      </StyledCheckBoxGroup>
    )
  },
)
