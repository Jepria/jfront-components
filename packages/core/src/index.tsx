export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
