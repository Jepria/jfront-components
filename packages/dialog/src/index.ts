export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'