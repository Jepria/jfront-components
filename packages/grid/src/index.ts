export * from './common'
export * from './jepgrid'
