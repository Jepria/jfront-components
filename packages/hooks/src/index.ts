export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
