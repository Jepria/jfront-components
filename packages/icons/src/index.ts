export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
