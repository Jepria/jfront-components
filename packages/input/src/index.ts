export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
