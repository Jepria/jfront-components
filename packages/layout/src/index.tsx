import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
