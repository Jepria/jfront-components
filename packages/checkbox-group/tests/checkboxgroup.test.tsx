import React from "react"
import { render, fireEvent, screen } from "@testing-library/react"
import { Simulate } from "react-dom/test-utils"
import { CheckBoxGroup } from "../src"
import { CheckBox } from "../../checkbox/src"

test("Checking for the existence of an element CheckBoxGroup", () => {
  expect(
    render(
      <CheckBoxGroup name="name" isLoading={false}>
        <CheckBox disabled={true} label="label1" value="value1" />
        <CheckBox label="label2" value="value2" />
        <CheckBox label="label3" value="value3" />
      </CheckBoxGroup>,
    ),
  )
})

test("Checking when clicking on CheckBox in CheckBoxGroup", () => {
  let pressed: any[] | undefined = []
  render(
    <CheckBoxGroup
      name="name"
      isLoading={false}
      onChange={(name, values) => {
        pressed = values
      }}
    >
      <CheckBox disabled={true} label="label1" value="value1" />
      <CheckBox label="label2" value="value2" />
      <CheckBox label="label3" value="value3" />
    </CheckBoxGroup>,
  )
  fireEvent.click(screen.getByLabelText("label1"))
  fireEvent.click(screen.getByLabelText("label2"))
  fireEvent.click(screen.getByLabelText("label1"))

  expect(pressed).toEqual(["value2"])
})

test("Matches snapshot ", () => {
  const { asFragment } = render(
    <CheckBoxGroup name="name" isLoading={false}>
      <CheckBox disabled={true} label="label1" value="value1" />
      <CheckBox label="label2" value="value2" />
      <CheckBox label="label3" value="value3" />
    </CheckBoxGroup>,
  )
  expect(asFragment()).toMatchSnapshot()
})
test("CheckBoxGroup isLoading renders correctly", () => {
  const tools = render(
    <div>
      <CheckBoxGroup name="name" disabled={false} isLoading>
        <CheckBox label="label1" value="value1" />
        <CheckBox label="label2" value="value2" />
      </CheckBoxGroup>
    </div>,
  )
  expect(tools.asFragment()).toMatchSnapshot()
})

test("CheckBoxGroup isError renders correctly", () => {
  const tools = render(
    <div>
      <CheckBoxGroup name="name" disabled={false} error="Wrong value">
        <CheckBox label="label1" value="value1" />
        <CheckBox label="label2" value="value2" />
      </CheckBoxGroup>
    </div>,
  )
  expect(tools.asFragment()).toMatchSnapshot()
})
