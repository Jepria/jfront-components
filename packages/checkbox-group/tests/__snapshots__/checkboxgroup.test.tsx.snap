// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup isError renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-AxiKw dVCmtU"
    >
      <div
        class="sc-AxhCb JzKgM"
        direction="column"
      >
        <span
          class="sc-AxheI ektFVT"
        >
          <label
            class="sc-AxgMl dzQsAn"
            for="id21"
          >
            label1
          </label>
          <input
            class="sc-AxhUy ifPyXr"
            id="id21"
            type="checkbox"
            value="value1"
          />
        </span>
        <span
          class="sc-AxheI ektFVT"
        >
          <label
            class="sc-AxgMl dzQsAn"
            for="id22"
          >
            label2
          </label>
          <input
            class="sc-AxhUy ifPyXr"
            id="id22"
            type="checkbox"
            value="value2"
          />
        </span>
      </div>
      <img
        class="sc-AxjAm khJDGy"
        src="data:image/gif;base64,R0lGODlhEAAQAIcAAED/QMJLMsNNNMVPNsNUPcxVPMZXQcddRs5dTNBZQcxiTM5kTtVgR9FjTNFhUNRkUthwWd5wWd1wXd18b919b+FdUuJfU+NhVOViVeZlVudnV+NmW+RnXOdsXehpWOpsWetuWulvX+F0XOVyWep2XexwW+5zXO92XfB4X/F6X+BzYO57Yel9aPN8YPN+YemCau2BaeyBbO6DbuOEdeaIeOqHfeuJfeqNe+yLf+ySf/KAZfSAYvWBYvWCY/SEZ/WEaveGbPaJa/iGbfONdvWOd/eMdfqOd/qUfN6hleWOg+eQhOeRheySiO2SieyYie2flfGTgvCXg/GXhPSZhPCfjvGdkOGkl+Onmu6glfihiPiiifijivOmkfSmkvGlmfGom/Gon/Kpn+uvovOvpvSsofWwo/ezoPezpPC0qPi0oPi2ofi1pfm5pvq8qPu9qfi+sfq/sv3DtfrZ0////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAMAAAAALAAAAAAQABAAAAjBAAEIHEiwoME4R4D8IPLG4EAjMrqkScOFBQyHQqS42RIkiJY2OUYULBLFTY8ec+a4aMHmhoiBcGKY5LEjZQoUJ9ZIECNwCJcsO1amPGGiBBQnEASuUOOjxc2UJUB8CPElgUASZnTgNCFnzgcPGjp4KSDwBZUpRUFUGaMhAwYbSxYIRKPijFQPKTFcsBDGgZWBEWiUaduEiYUKYSgYKMhgBhkcHDbUADMhgMMGD5RgeZIEAQGHAq8oGCDgABLQqAkGBAA7"
        title="Wrong value"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`CheckBoxGroup isLoading renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-AxiKw dIjHRK"
    >
      <div
        class="sc-AxhCb JzKgM"
        direction="column"
      >
        <span
          class="sc-AxheI ektFVT"
        >
          <label
            class="sc-AxgMl dzQsAn"
            for="id19"
          >
            label1
          </label>
          <input
            class="sc-AxhUy ifPyXr"
            id="id19"
            type="checkbox"
            value="value1"
          />
        </span>
        <span
          class="sc-AxheI ektFVT"
        >
          <label
            class="sc-AxgMl dzQsAn"
            for="id20"
          >
            label2
          </label>
          <input
            class="sc-AxhUy ifPyXr"
            id="id20"
            type="checkbox"
            value="value2"
          />
        </span>
      </div>
      <img
        class="sc-AxirZ cxtyjb"
        src="data:image/gif;base64,R0lGODlhEAAQALMMAKqooJGOhp2bk7e1rZ2bkre1rJCPhqqon8PBudDOxXd1bISCef///wAAAAAAAAAAACH/C05FVFNDQVBFMi4wAwEAAAAh+QQFAAAMACwAAAAAEAAQAAAET5DJyYyhmAZ7sxQEs1nMsmACGJKmSaVEOLXnK1PuBADepCiMg/DQ+/2GRI8RKOxJfpTCIJNIYArS6aRajWYZCASDa41Ow+Fx2YMWOyfpTAQAIfkEBQAADAAsAAAAABAAEAAABE6QyckEoZgKe7MEQMUxhoEd6FFdQWlOqTq15SlT9VQM3rQsjMKO5/n9hANixgjc9SQ/CgKRUSgw0ynFapVmGYkEg3v1gsPibg8tfk7CnggAIfkEBQAADAAsAAAAABAAEAAABE2QycnOoZjaA/IsRWV1goCBoMiUJTW8A0XMBPZmM4Ug3hQEjN2uZygahDyP0RBMEpmTRCKzWGCkUkq1SsFOFQrG1tr9gsPc3jnco4A9EQAh+QQFAAAMACwAAAAAEAAQAAAETpDJyUqhmFqbJ0LMIA7McWDfF5LmAVApOLUvLFMmlSTdJAiM3a73+wl5HYKSEET2lBSFIhMIYKRSimFriGIZiwWD2/WCw+Jt7xxeU9qZCAAh+QQFAAAMACwAAAAAEAAQAAAETZDJyRCimFqbZ0rVxgwF9n3hSJbeSQ2rCWIkpSjddBzMfee7nQ/XCfJ+OQYAQFksMgQBxumkEKLSCfVpMDCugqyW2w18xZmuwZycdDsRACH5BAUAAAwALAAAAAAQABAAAARNkMnJUqKYWpunUtXGIAj2feFIlt5JrWybkdSydNNQMLaND7pC79YBFnY+HENHMRgyhwPGaQhQotGm00oQMLBSLYPQ9QIASrLAq5x0OxEAIfkEBQAADAAsAAAAABAAEAAABE2QycmUopham+da1cYkCfZ94UiW3kmtbJuRlGF0E4Iwto3rut6tA9wFAjiJjkIgZAYDTLNJgUIpgqyAcTgwCuACJssAdL3gpLmbpLAzEQA7"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw dIjHRK"
  >
    <div
      class="sc-AxhCb JzKgM"
      direction="column"
    >
      <span
        class="sc-AxheI ektFVT"
      >
        <label
          class="sc-AxgMl dzQsAn"
          for="id16"
        >
          label1
        </label>
        <input
          class="sc-AxhUy ifPyXr"
          disabled=""
          id="id16"
          type="checkbox"
          value="value1"
        />
      </span>
      <span
        class="sc-AxheI ektFVT"
      >
        <label
          class="sc-AxgMl dzQsAn"
          for="id17"
        >
          label2
        </label>
        <input
          class="sc-AxhUy ifPyXr"
          id="id17"
          type="checkbox"
          value="value2"
        />
      </span>
      <span
        class="sc-AxheI ektFVT"
      >
        <label
          class="sc-AxgMl dzQsAn"
          for="id18"
        >
          label3
        </label>
        <input
          class="sc-AxhUy ifPyXr"
          id="id18"
          type="checkbox"
          value="value3"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
